modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/io/DDMFormValuesJSONDeserializer.java

```diff
@@ -136,7 +136,14 @@ protected DDMFormFieldValue getDDMFormFieldValue(
 
 		ddmFormFieldValue.setFieldReference(
 			jsonObject.getString("fieldReference"));
-		ddmFormFieldValue.setInstanceId(jsonObject.getString("instanceId"));
+
+		String instanceId = jsonObject.getString("instanceId");
+
+		if (!instanceId.matches("[a-zA-Z0-9]*")) {
+			return null;
+		}
+
+		ddmFormFieldValue.setInstanceId(instanceId);
```

Comment:
@natocesarrego I think that sounds fine, but it would break the test that we just added.

@binhtran92 Would you have time to update the code to look like this, and then fix the test?

```java
		if (instanceId.matches("[a-zA-Z0-9]*")) {
			ddmFormFieldValue.setInstanceId(instanceId);
		}
```

Reply:
@binhtran92 Never mind, I had some time to rewrite the test.

@natocesarrego I've applied the recommended update, and I've separated the test commit from the original commit to make it easier to see the test failure prior to the changes.